packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts

```diff
@@ -331,49 +331,34 @@ describe('portfolio associations and product constraints', () => {
     Template.fromStack(stack).resourceCountIs('AWS::ServiceCatalog::TagOptionAssociation', 3);
   }),
 
-  test('adding identical tag options to portfolio is idempotent', () => {
-    const tagOptions1 = new servicecatalog.TagOptions({
-      key1: ['value1', 'value2'],
-      key2: ['value1'],
-    });
-
-    const tagOptions2 = new servicecatalog.TagOptions({
-      key1: ['value1', 'value2'],
-    });
-
-    portfolio.associateTagOptions(tagOptions1);
-    portfolio.associateTagOptions(tagOptions2); // If not idempotent this would fail
-
-    Template.fromStack(stack).resourceCountIs('AWS::ServiceCatalog::TagOption', 3); //Generates a resource for each unique key-value pair
-    Template.fromStack(stack).resourceCountIs('AWS::ServiceCatalog::TagOptionAssociation', 3);
-  }),
-
   test('fails to add tag options with invalid minimum key length', () => {
```

Comment:
These are also not really valid but we can modify it slightly to still work, unfortunately still a lot of git churn.  Moving the creation away from the association means that early failures happen on creation not via association calls.

Reply:
updated with modified wording and minimized changes.